packages/jaeger-ui/src/components/TracePage/TraceTimelineViewer/SpanTreeOffset.js

```diff
@@ -52,14 +52,7 @@ export class UnconnectedSpanTreeOffset extends React.PureComponent<SpanTreeOffse
     super(props);
 
     this.ancestorIds = [];
-    let currentSpan: Span = props.span;
-    while (currentSpan) {
-      currentSpan = _get(_find(currentSpan.references, { refType: 'CHILD_OF' }), 'span');
-      if (currentSpan) {
-        this.ancestorIds.push(currentSpan.spanID);
-      }
-    }
-
```

Comment:
Looking at the code that builds the trace tree, which is then turned into the array of spans with parents preceding children (and children are sorted by startTime):

https://github.com/jaegertracing/jaeger-ui/blob/38efa49e96d34cb666fa232da14346ceccd896d4/packages/jaeger-ui/src/selectors/trace.js#L72-L75

Looks like the algorithm (which could probably be better) is to use the span ID of the first reference, as long as that reference is FOLLOWS_FROM or CHILD_OF.

In accumulating the ancestor IDs, we should aim to be consistent with that process so it accurately reflects the order the list.

Therefore, maybe the following would work?

```jsx
while (currentSpan && Array.isArray(currentSpan.references)) {
  const { (refType, span } = currentSpan.references[0] || {};
  if (span && (refType === 'FOLLOWS_FROM' || refType === 'CHILD_OF')) {
    this.ancestorIds.push(span.spanID);
    currentSpan = span;
  }
}
```

What do you think? 

@everett980 Can you weigh in?

Reply:
I think the precedent for handling reference is not particularly strong. AFAIK there is nothing preventing a span from being `FOLLOWS_FROM` multiple other spans. With regards to indentation it should only indent once (based on depth) but I do not know how the UI handles this case in other ways. Does the span with two `FOLLOWS_FROM` show up twice in the list, once per reference?

Purely from a UI layout perspective I like the implementation in your comment. However, I am uncertain if we should have better support for multiple references.

Reply:
Multiple references are rarely used within the same trace (i.e. to represent a JOIN), the more realistic use case is something like multiple traces coalescing into a single batch write operation, which causally depends on all of the previous traces. However, the _first_ reference in the list is always used to pick the trace ID for the new span (actually, different Jaeger clients might be different in that regard, e.g.. they may give priority to the _first child-of reference_).